com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/Raytracing/HDTemporalFilter.RenderGraph.cs

```diff
@@ -0,0 +1,164 @@
+using UnityEngine.Experimental.Rendering;
+using UnityEngine.Experimental.Rendering.RenderGraphModule;
+using UnityEngine.Rendering;
+
+namespace UnityEngine.Rendering.HighDefinition
+{
+    partial class HDTemporalFilter
+    {
+        class TemporalFilterPassData
+        {
+            public TemporalFilterParameters parameters;
+            public TextureHandle depthStencilBuffer;
+            public TextureHandle normalBuffer;
+            public TextureHandle motionVectorBuffer;
+            public TextureHandle velocityBuffer;
+            public TextureHandle historyDepthTexture;
+            public TextureHandle historyNormalTexture;
+            public TextureHandle noisyBuffer;
+            public TextureHandle validationBuffer;
+            public TextureHandle historyBuffer;
+            public TextureHandle outputBuffer;
+        }
+
+        public TextureHandle Denoise(RenderGraph renderGraph, HDCamera hdCamera, TemporalFilterParameters tfParameters, TextureHandle noisyBuffer, TextureHandle historyBuffer, TextureHandle depthPyramid, TextureHandle normalBuffer, TextureHandle motionVectorBuffer)
+        {
+            using (var builder = renderGraph.AddRenderPass<TemporalFilterPassData>("TemporalDenoiser", out var passData, ProfilingSampler.Get(HDProfileId.TemporalFilter)))
+            {
+                // Cannot run in async
+                builder.EnableAsyncCompute(false);
+
+                // Fetch all the resources
+                passData.parameters = tfParameters;
+                passData.depthStencilBuffer = builder.ReadTexture(depthPyramid);
+                passData.normalBuffer = builder.ReadTexture(normalBuffer);
+                passData.motionVectorBuffer = builder.ReadTexture(motionVectorBuffer);
+                passData.velocityBuffer = renderGraph.defaultResources.blackTextureXR;
+                passData.historyDepthTexture = renderGraph.ImportTexture(hdCamera.GetCurrentFrameRT((int)HDCameraFrameHistoryType.Depth));
```

Comment:
Missing read/write info on imported buffers here. It won't have an impact since imported buffer's lifetime is not handled by rendergraph but it can be used for debug display purpose so it's better to be consistent and specify it for imported buffers as well.